### PR TITLE
Devex dry up handlers

### PR DESCRIPTION
**Pull request by matthopson**
Cleans up the lambda functions by abstracting the handler logic. Additionally, the `handle` function is modified to look for pdf data so the `customHandle` function could go away completely.

### CODE REVIEW

File: web-api/src/genericHandler.test.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+    expect(logged.includes('User')).toBeFalsy();
+    expect(logged.includes('Results')).toBeFalsy();
+    expect(logged.includes('Event')).toBeFalsy();
+
+    expect(logged.includes('The thing that happened')).toBeTruthy();
+    expect(logged.includes('The stuff that came out')).toBeTruthy();
+    expect(logged.includes('Who did the thing')).toBeTruthy();
+  });
+
+  it('returns the results of a successful exectution', async () => {
```
Comment (rachaelparris) [inline web-api/src/genericHandler.test.js]: `execution`

File: web-api/src/documents/getDocumentDownloadUrlLambda.js [hunk 2]
```diff
@@ -11,22 +7,12 @@ const {
...
-          documentId: event.pathParameters.documentId,
-        });
-      applicationContext.logger.info('User', user);
-      applicationContext.logger.info('Results', results);
-      return results;
-    } catch (e) {
-      applicationContext.logger.error(e);
-      throw e;
-    }
+  genericHandler(event, async ({ applicationContext }) => {
```
Comment (rachaelparris) [inline web-api/src/documents/getDocumentDownloadUrlLambda.js]: As discussed, switch back to use `redirect`.

File: web-api/src/public-api/getPublicDocumentDownloadUrlLambda.js [hunk 3]
```diff
@@ -8,20 +7,19 @@ const { redirect } = require('../middleware/apiGatewayHelper');
  * @returns {Promise<*|undefined>} the api gateway response object containing the statusCode, body, and headers
  */
 exports.handler = event =>
-  redirect(event, async () => {
-    const applicationContext = createApplicationContext({});
-    try {
-      const results = await applicationContext
+  genericHandler(
```
Comment (rachaelparris) [inline web-api/src/public-api/getPublicDocumentDownloadUrlLambda.js]: Same as above, switch back to use `redirect`.

**Review by rachaelparris — APPROVED** [review]
Other than the few things mentioned above, 👍. Awesome!

**Comment by matthopson** [comment]
@rachaelparris I'm going to leave customHandle in for now and just keep an eye on this in case we need to put it back. Setting myself a reminder to remove it next week.

**Review by kkoskelin — APPROVED** [review]
Almost as good as sliced bread.

Homemade bread at 9000ft elevation.

**Pull request merged**